extensions/ql-vscode/src/query-server/query-server-client.ts

```diff
@@ -125,6 +125,7 @@ export class QueryServerClient extends DisposableObject {
           this.restartQueryServerInternal(progress, token),
         {
           title: "Restarting CodeQL query server due to unexpected termination",
+          cancellable: true,
```

Comment:
I don't think this should be cancellable either for the same reason as the other query server restart.